drivers/input/input_handlers.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022, tangchunhui@coros.com
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/input.h>
+#include <zephyr/syscall_handler.h>
+
+static inline int z_vrfy_input_setup(const struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_INPUT(dev, setup));
+
+	return z_impl_input_setup((const struct device *)dev);
+}
+#include <syscalls/input_setup_mrsh.c>
+
+static inline int z_vrfy_input_release(const struct device *dev)
+{
+	Z_OOPS(Z_SYSCALL_DRIVER_INPUT(dev, release));
+
+	return z_impl_input_release((const struct device *)dev);
+}
+#include <syscalls/input_release_mrsh.c>
+
+static inline int z_vrfy_input_config_get(const struct device *dev, struct input_config *config)
+{
+	struct input_config _config;
+	int retval = 0;
+
+	Z_OOPS(Z_SYSCALL_DRIVER_INPUT(dev, config_get));
+
+	retval = z_impl_input_config_get((const struct device *)dev, &_config);
+	Z_OOPS(z_user_to_copy(config, &_config, sizeof(struct input_config)));
+
+	return retval;
+}
+#include <syscalls/input_config_get_mrsh.c>
+
+static inline int z_vrfy_input_config_set(const struct device *dev, struct input_config *config)
+{
+	struct input_config _config;
+
+	Z_OOPS(Z_SYSCALL_DRIVER_INPUT(dev, config_set));
+	Z_OOPS(z_user_from_copy(&_config, config, sizeof(struct input_config)));
+
+	return z_impl_input_config_set((const struct device *)dev, _config);
+}
+#include <syscalls/input_config_set_mrsh.c>
+
+static inline int z_vrfy_input_event_read(const struct device *dev, struct input_event *event)
```

Comment:
async api would be very useful
blocking api forces you to use a dedicated thread to handle read.

I suggest creating a pollable sync object that can be used similarly as in linux poll.
Here is the example from NCS how it can be done https://github.com/nrfconnect/sdk-nrf/blob/main/samples/nfc/tag_reader/src/main.c#L703